switchbot/__init__.py

```diff
@@ -220,42 +313,47 @@ def press(self) -> bool:
         return self._sendcommand(PRESS_KEY, self._retry_count)
 
     def switch_mode(self) -> str:
-        """Return Toggle or Switch from cache.
-        Run update first."""
-        return self._mode
+        """Return true or false from cache."""
+        # To get actual position call update() first.
+        if not self._switchbot_device_data:
+            self.update()
+        return self._switchbot_device_data["data"]["switchMode"]
 
     def is_on(self) -> bool:
-        """Return switch state from cache.
-        Run update first."""
-        return self._is_on
+        """Return switch state from cache."""
+        # To get actual position call update() first.
+        if not self._switchbot_device_data:
+            self.update()
```

Comment:
```suggestion
            return None
```
Do not want is_on to trigger any scan

Reply:
Done